### PR TITLE
Making installer tests pass on OSX-arm64 in Release build

### PR DESCRIPTION
**Pull request by VSadov**
These are tests/infra changes mostly dealing with codesign issues that crept up over time.

It is enough to pass tests in Release
As in `./build.sh clr+libs+host+packs+host.tests -c Release -rc Release -lc Release -test`

### CODE REVIEW

**Comment by VSadov** [comment]
Sadly, this is not enough to pass with Checked runtime. 
(as in `./build.sh clr+libs+host+packs+host.tests -c Release -rc Checked -lc Release -test` )

It appears that some of the code patterns around `pal::string_t` trigger fatal crashes on deallocation paths. I am still not sure if those are real issues or OSX allocator is overly protective in Checked. 

Either way that will require changes in the host code (vs. just tests/infra changes).

File: src/installer/tests/HostActivation.Tests/PortableAppActivation.cs [hunk 1]
```diff
@@ -553,7 +554,7 @@ public SharedTestState()
                 MockApp = new TestApp(SharedFramework.CalculateUniqueTestDirectory(Path.Combine(TestArtifact.TestArtifactsPath, "portableAppActivation")), "App");
                 Directory.CreateDirectory(MockApp.Location);
                 File.WriteAllText(MockApp.AppDll, string.Empty);
-                MockApp.CreateAppHost(copyResources: false);
+                MockApp.CreateAppHost(copyResources: false, enableMacOSCodeSign: true);
```
Comment (vitek-karas) [inline src/installer/tests/HostActivation.Tests/PortableAppActivation.cs:557]: Isn't the default value `true` already? (same for the other places as well).

File: src/installer/tests/HostActivation.Tests/FrameworkResolution/MultipleHives.cs [hunk 2]
```diff
@@ -137,6 +139,7 @@ private List<FrameworkInfo> GetExpectedFrameworks(bool? multiLevelLookup)
             return expectedList;
         }
 
+        [SkipOnPlatform(TestPlatforms.OSX, "https://github.com/dotnet/runtime/issues/91486")]
```
Comment (elinor-fung) [inline src/installer/tests/HostActivation.Tests/FrameworkResolution/MultipleHives.cs:142]: Can we narrow this to osx arm64? These should be running (and passing) on osx x64 CI builds currently.

File: src/installer/tests/HostActivation.Tests/DependencyResolution/RidAssetResolution.cs [hunk 3]
```diff
@@ -139,6 +139,7 @@ public void RidSpecificAssembly_RidGraph(string rid, string includedPath, string
                 excludedPath);
         }
 
+        [SkipOnPlatform(TestPlatforms.OSX, "https://github.com/dotnet/runtime/issues/91486")]
```
Comment (elinor-fung) [inline src/installer/tests/HostActivation.Tests/DependencyResolution/RidAssetResolution.cs:142]: I'm missing why the two tests in here are affected by this issue. They don't use `TestOnlyProductBehavior`?

**Pull request closed without merging**